src/Domain/Entity/PurchaseEntity.cs

```diff
@@ -0,0 +1,207 @@
+
+    [Required(ErrorMessage = ErrorMessages.PurchaseEntityPurchaseLinesRequired)]
+    public ICollection<PurchaseLineItem> Purchases
+    {
+        get => _purchases;
+        set
+        {
+            ValidatePurchases(value);
+            _purchases = value;
+        }
+    }
+
+    private double? _vatAmount;
+
+    [Range(0, double.MaxValue, ErrorMessage = ErrorMessages.PurchaseEntityVatAmountPositive)]
+    public double? VatAmount
+    {
+        get => _vatAmount;
+        set
+        {
+            ValidateVatAmount(value);
+            _vatAmount = value;
+        }
+    }
+
+    private double? _transportFee;
+    [Range(0, double.MaxValue, ErrorMessage = ErrorMessages.PurchaseEntityTransportFeePositive)]
+    public double? TransportFee
+    {
+        get => _transportFee;
+        set
+        {
+            ValidateTransportFee(value);
+            _transportFee = value;
+        }
+    }
+    
+    private double? _paidAmount;
+    [Range(0, double.MaxValue, ErrorMessage = ErrorMessages.PurchaseEntityPaidAmountPositive)]
+    public double? PaidAmount
+    {
+        get => _paidAmount;
+        set
+        {
+            ValidatePaidAmount(value);
+            _paidAmount = value;
+        }
+    }
+    
+    private string? _remarks;
+    [MaxLength(500, ErrorMessage = ErrorMessages.PurchaseEntityRemarksTooLong)]
+    public string? Remarks
+    {
+        get => _remarks;
+        set
+        {
+            ValidateRemarks(value);
+            _remarks = value;
+        }
+    }
+
+    private void ValidateRemarks(string? value)
+    {
+        if (value != null && value.Length > 500)
+        {
+            throw new DomainValidationException("Remarks", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityRemarksTooLong);
+        }
+    }
+
+    private int? _invoiceNumber;
+
+    [Range(0, int.MaxValue, ErrorMessage = ErrorMessages.PurchaseEntityInvoiceNumberPositive)]
+    public int? InvoiceNumber
+    {
+        get => _invoiceNumber;
+        set
+        {
+            ValidateInvoiceNumber(value);
+            _invoiceNumber = value;
+        }
+    }
+
+    private DateOnly _date;
+
+    [Required(ErrorMessage = ErrorMessages.PurchaseEntityDateRequired)]
+    public DateOnly Date
+    {
+        get => _date;
+        set
+        {
+            ValidateDate(value);
+            _date = value;
+        }
+    }
+    
+    private void ValidatePurchases(ICollection<PurchaseLineItem> value)
+    {
+        foreach (var item in value)
+        {
+            var validationResults = new List<ValidationResult>();
+            var context = new ValidationContext(item);
+            if (!Validator.TryValidateObject(item, context, validationResults, true))
+            {
+                throw new DomainValidationException("PurchaseLineItem", ErrorCode.BadRequest,
+                    ErrorMessages.PurchaseEntityInvalidPurchaseLineItem);
+            }
+        }
+    }
```

Comment:
Okay, maybe you need to explain whats going on here. What is ValidationContext ??

Reply:
I now realize what you are doing here. 
But the sole fact that lineItems EXISTS MEANS they are valid. 
That is the whole idea of domain driven design.

If they were invalid, that would have already thrown exception where they were created.